src/s3path.jl

```diff
@@ -107,7 +111,8 @@ function Base.tryparse(::Type{<:S3Path}, str::AbstractString; config::Union{Noth
     path = ()
     isdirectory = true
 
-    tokenized = split(str, "/")
+    version_split = split(str, "?versionId=")
+    tokenized = split(version_split[1], "/")
     bucket = strip(tokenized[3], '/')
     drive = "s3://$bucket"
```

Comment:
Let's avoid re-inventing the wheel here and make use of URIs.jl:
```suggestion
    uri = URI(str)
    params = queryparams(uri)
    version = get(params, "versionId", nothing)
    bucket = uri.host
    drive = "s3://$bucket"
    tokenized = [uri.scheme, "", uri.host, split(lstrip(uri.path, '/'), '/')...]
```
Note you could clean up `tokenized` more as it just seems to care about `uri.path` after this point

Reply:
Done! I added the new testset first, with test cases based on the old behavior, then updated the behavior in a separate commit and made sure they still passed.